backend/cmm_helpers.mli

```diff
@@ -1117,9 +1147,16 @@ val unboxed_mutable_int32_unboxed_product_array_set :
     The [has_custom_ops] parameter should be set to [true] unless the array
     in question is an unboxed product array: these are represented as mixed
     blocks, not custom blocks.
+
+    The zero-indexed element number is specified as a tagged immediate.
+    This coincides with the index in words iff [has_custom_ops] is [false].
```

Comment:
I don't think this last sentence is necessary, or if you want to keep it, then you should also specify what happens when `[has_custom_ops] is [true]`.

Reply:
Maybe also leave a CR for swapping arrays of unboxed scalars to the mixed block representation